editor/utils/graphql.ts

```diff
@@ -100,20 +100,20 @@ export const useQuery = <T extends Record<string, unknown>>({
   hasAggregates,
 }: UseQueryProps): UseQueryResponse<T> => {
   const { user } = useAuth0();
-  const token = useToken();
+  const getToken = useGetToken();
 
   const fetchWithAuth = async ([query, variables]: [
     RequestDocument,
     Variables,
   ]): Promise<HasuraGraphQLResponse<T, typeof typename>> => {
     const hasuraRoleName = getHasuraRoleName(getRolesArray(user));
+    const token = await getToken();
```

Comment:
again, here you can see the `getToken()` called on demand, right before we execute the request.